source/extensions/transport_sockets/alts/noop_transport_socket_callbacks.h

```diff
@@ -0,0 +1,35 @@
+#include "envoy/network/transport_socket.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+
+/**
+ * A TransportSocketCallbacks for wrapped TransportSocket object. Some
+ * TransportSocket implementation wraps another socket which does actual I/O.
+ * This class is used by the wrapped socket as its callbacks instead of the real
+ * connection to hold back callbacks from the underlying socket to connection.
+ */
+class NoOpTransportSocketCallbacks : public Network::TransportSocketCallbacks {
+public:
+  explicit NoOpTransportSocketCallbacks(Network::TransportSocketCallbacks& parent)
+      : parent_(parent) {}
+
+  int fd() const override { return parent_.fd(); }
+  Network::Connection& connection() override { return parent_.connection(); }
+  bool shouldDrainReadBuffer() override { return false; }
+  /*
+   * No-op for these two methods to hold back the callbacks.
+   */
+  void setReadBufferReady() override {}
+  void raiseEvent(Network::ConnectionEvent) override {}
```

Comment:
I'm fine with this landing as is, but can you please loop me in when it is used?  I have some concerns about these particular events being swallowed.  Maybe we can add ASSERTs that they're not used so we'll at least catch it in debug builds if a mistake is made?  Hopefully seeing where the code is used will allay my concerns :-)

Reply:
ACK'ed

Reply:
@alyssawilk I will merge master and use this in my PR #4153, ASSERT cannot be added here. In #4153, it have to swallow `Connected` event from RawBufferSocket, and raise it when handshake is done.